jOOQ/src/main/java/org/jooq/impl/DefaultRecordMapper.java

```diff
@@ -981,6 +1014,44 @@ public final E map(R record) {
         }
     }
 
+    private class KotlinDataClassMapper implements RecordMapper<R, E> {
```

Comment:
Is this class really necessary? Can't we use the existing `ImmutablePOJOMapperWithParameterNames`?